cmd/podman/system/connection/add.go

```diff
@@ -89,30 +94,20 @@ func add(cmd *cobra.Command, args []string) error {
 		uri.Path = cmd.Flag("socket-path").Value.String()
 	}
 
-	switch uri.Scheme {
-	case "ssh":
-		if uri.User.Username() == "" {
-			if uri.User, err = utils.GetUserInfo(uri); err != nil {
-				return err
-			}
-		}
+	var sshMode sshCommon.EngineMode
+	containerConfig := registry.PodmanConfig()
 
-		if cmd.Flags().Changed("port") {
-			uri.Host = net.JoinHostPort(uri.Hostname(), cmd.Flag("port").Value.String())
-		}
+	flag := containerConfig.SSHMode
+	switch flag {
+	case "native":
+		sshMode = sshCommon.NativeMode
+	case "golang":
+		sshMode = sshCommon.GolangMode
+	}
```

Comment:
The parsing code should probably be moved into the `ssh` package.  That'll prevent callers to diverge from the callee.

Reply:
ok